src/agent/src/mount.rs

```diff
@@ -1012,6 +1015,40 @@ fn ensure_destination_file_exists(path: &Path) -> Result<()> {
     Ok(())
 }
 
+/// Parse mtab and return the device which is mounted at a given directory
+pub fn get_mount_device(mount_dir: &Path) -> Result<String> {
+    let dir = mount_dir.to_string_lossy().into_owned();
+    let f = File::open(MTAB_PATH)?;
+    let reader = BufReader::new(f);
+
+    for line in reader.lines() {
+        let tem_line = line?;
+        let parts: Vec<&str> = tem_line.split_whitespace().collect();
+        if parts.contains(&dir.as_str()) && !parts.is_empty() {
+            return Ok(String::from(parts[0]));
+        }
+    }
+
+    return Err(anyhow!("can't find mount point {:?}", dir));
+}
+
+#[instrument]
+pub fn expand_file_system(mountpoint: &String, logger: Logger) -> Result<()> {
+    let path = Path::new(mountpoint);
```

Comment:
`get_mount_device` accepts a `Path` parameter, here you convert a `String` to `Path` and in `get_mount_device` you convert it to `String` back, so why not just use a `String` or `&str` parameter for `get_mount_device` fn?